Lib/profiling/sampling/heatmap_index_template.html

```diff
@@ -0,0 +1,81 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Tachyon Profiler - Heatmap Report</title>
+    <!-- INLINE_CSS -->
+</head>
+<body>
+    <div class="page-wrapper">
+        <div class="container">
+            <header class="header">
+                <div class="header-content">
+                    <!-- PYTHON_LOGO -->
+                    <div class="header-text">
+                        <h1>Tachyon Profiler Heatmap Report</h1>
+                        <p class="subtitle">Line-by-line performance analysis</p>
+                    </div>
+                </div>
+            </header>
+
+            <div class="stats-summary">
+                <div class="stat-card">
+                    <span class="stat-value"><!-- NUM_FILES --></span>
+                    <span class="stat-label">Files Profiled</span>
+                </div>
+                <div class="stat-card">
+                    <span class="stat-value"><!-- TOTAL_SAMPLES --></span>
+                    <span class="stat-label">Total Samples</span>
+                </div>
+                <div class="stat-card">
+                    <span class="stat-value"><!-- DURATION --></span>
+                    <span class="stat-label">Duration</span>
+                </div>
+                <div class="stat-card">
+                    <span class="stat-value"><!-- SAMPLE_RATE --></span>
+                    <span class="stat-label">Samples/sec</span>
+                </div>
+            </div>
+
+            <div class="file-list">
+                <h2>Profiled Files</h2>
+
+                <div class="filter-controls">
+                    <input type="text" id="searchInput" placeholder="🔍 Search files..." onkeyup="filterTable()">
+                    <select id="moduleFilter" onchange="filterTable()">
+                        <option value="all">All Modules</option>
+                        <option value="stdlib">Standard Library</option>
+                        <option value="site-packages">Site Packages</option>
+                        <option value="project">Project</option>
+                        <option value="other">Other</option>
+                    </select>
+                </div>
+
+                <table id="fileTable">
+                    <thead>
+                        <tr>
+                            <th onclick="sortTable(0)">File ⇅</th>
+                            <th onclick="sortTable(1)">Module ⇅</th>
+                            <th onclick="sortTable(2)">Type ⇅</th>
+                            <th onclick="sortTable(3)" style="text-align: right;">Samples ⇅</th>
+                            <th onclick="sortTable(4)" style="text-align: right;">Lines Hit ⇅</th>
+                            <th onclick="sortTable(5)" style="text-align: right;">% of Total ⇅</th>
+                            <th style="text-align: left;">Intensity</th>
```

Comment:
I don't know if I was sleepy or what but I genuinely had to use brainpower ™️ to understand these columns.

IIUC, these two columns show the same data in different formats. It was not initially obvious to me that % of total wasn't intended to sum to 100% and that the % and bar are actually relative to the hottest file (which should always equal 100%). I feel like my brain expected the bars below the top file to be a proportion of a whole.

A thought here might be to merge the columns into a single column called "Intensity" and then show the percentage number with a coloured background using the same blue --> red heatmap. We could also add a note in the header that this is really the % relative to the hottest file.